sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -215,7 +215,7 @@ case class ExpressionEncoder[T](
    */
   def assertUnresolved(): Unit = {
     (fromRowExpression +:  toRowExpressions).foreach(_.foreach {
-      case a: AttributeReference =>
+      case a: AttributeReference if a.name != "loopVar" =>
```

Comment:
Hmm, this does feel kind of hacky, but I'm not sure if its really a problem in practice.  This assertion is only to catch our own bugs, not user mistakes.